### PR TITLE
Add `skypilot` support for AWS and GCP

### PR DESCRIPTION
**Pull request by strickvl**
Adds support for the skypilot orchestrator (https://github.com/zenml-io/zenml/pull/1765/) to AWS and GCP modular recipes.

Ideally would also have [this ticket](https://zenml.atlassian.net/browse/OSS-2241) addressed, but in the meanwhile we can at least partially support this by allowing for the zenml server, artifact store and container registry to be deployed and registered alongside the skypilot orchestrator in a single stack.

### CODE REVIEW

File: docs/book/stacks/k3d.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ as can a list of components that are coming soon.
 | Artifact Store     | minio                                   |
 | Container Registry | default                                 |
 | Experiment Tracker | mlflow                                  |
-| Orchestrator       | kubernetes, kubeflow, tekton, sagemaker |
+| Orchestrator       | kubeflow, kubernetes, sagemaker, tekton |
```
Comment (fa9r) [inline docs/book/stacks/k3d.md:17]: no skypilot here?

File: src/mlstacks/terraform/k3d-modular/variables.tf [hunk 2]
```diff
@@ -7,6 +7,10 @@ variable "enable_orchestrator_kubernetes" {
   description = "Enable Kubernetes deployment"
   default     = false
 }
+variable "enable_orchestrator_skypilot" {
+  description = "Enable SkyPilot orchestrator deployment"
+  default     = false
+}
```
Comment (fa9r) [inline src/mlstacks/terraform/k3d-modular/variables.tf:13]: Isn't skypilot cloud only? How would it relate to k3d-modular?

**Review by fa9r — APPROVED** [review]
Two minor questions otherwise looks good 👍

File: src/mlstacks/terraform/k3d-modular/variables.tf (hunk 2 above)
Comment (strickvl) [reply src/mlstacks/terraform/k3d-modular/variables.tf:13]: Good point.

**Pull request merged**